File: azCheck.js
//az check
function azHelper(str){
	if(str.indexOf("a") === -1 || str.indexOf("z") === -1 || str.indexOf("z") < str.indexOf("a")){
		return false;
		}
	return true;
}

function az(str){
	var idxA = str.indexOf("a");
	var idxZ = str.indexOf("z");
	var dist = idxA+3;
	
	if(azHelper(str) === true){
		if(idxZ <= dist){
			return true;
		}
	}
	return false;
}
//az("za");

File: az_check.rb
#az check
def az_helper(str)
	if str.index("z") == nil || str.index("a") == nil ||
		str.index("z") < str.index("a")
		return false
	else
		return true
	end
end

def az(str)
	indexA = str.index("a")
	indexZ = str.index("z")
	if indexA == nil		
		return false
	end
	if indexZ == nil
		return false
	end
	
	dist = indexA + 3
	if az_helper(str) == true
		if indexZ <= dist
			return true
		end
	return false
	end
end

az("abcz")